Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/Wrappers/SpeckleDataObjectWrapper.cs

```diff
@@ -10,10 +9,28 @@ namespace Speckle.Connectors.GrasshopperShared.Parameters;
 /// </summary>
 public class SpeckleDataObjectWrapper : SpeckleWrapper, ISpeckleCollectionObject
 {
+  /// <summary>
+  /// Name on the wrapper and wrapped Base (DataObject) are kept in sync here.
+  /// DataObject.name is the single source of truth - all geometry names inherit from it.
+  /// </summary>
+  public override string Name
+  {
+    get
+    {
+      SyncGeometriesToDataObject();
+      return DataObject.name;
+    }
+    set
+    {
+      DataObject.name = value;
+      SyncGeometriesToDataObject();
```

Comment:
I would only sync the name here, not name and props, since this is the name setter

Reply:
Split sync into targeted methods - Now `SyncGeometryNames()` and `SyncGeometryProperties()` handle their specific responsibilities